src/main/java/com/example/BookService/Controller/BookController.java

```diff
@@ -0,0 +1,55 @@
+package com.example.BookService.Controller;
+
+import com.example.BookService.Entity.Book;
+import com.example.BookService.Service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/books")
+public class BookController {
+
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping
+    public ResponseEntity<List<Book>> getAllBooks() {
+        List<Book> books = bookService.getAllBooks();
+        return ResponseEntity.ok(books);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> getBookById(@PathVariable Long id) {
```

Comment:
It is not good practice to return the entity, instead create dto (data transfer object) of its type. Why? https://saranganjana.medium.com/data-transfer-object-dto-in-java-66c4f3075362